### PR TITLE
Added checks for relative input shapes in linear regression using the utility introduced in #2370

### PR DESCRIPTION
**Pull request by lumi232**
This PR implements the solution mentioned in the issue https://github.com/mlpack/mlpack/issues/2820 by using the CheckSameSizes and CheckSameDimensionality utilities in src/mlpack/methods/linear_regression/linear_regression_train_main.cpp
src/mlpack/methods/linear_regression/linear_regression_main.cpp and 
src/mlpack/methods/linear_regression/linear_regression_predict_main.cpp
All the if statements mentioned in https://github.com/mlpack/mlpack/pull/2370#pullrequestreview-436211692 were replaced with the utility functions. 

I plan to implement the solution to other methods as well once this PR is merged successfully. This is my first PR I look forward to contribute more to the organisation. Looking forward for the review.

### CODE REVIEW

**Review by rcurtin — COMMENTED** [review]
Thanks for working on this @lumi232.  Things look good to me; I think that all the builds will pass (don't mind the Windows build failure, it is unrelated).  If you can handle the comments, I think we can get this merged. :+1:

File: src/mlpack/methods/linear_regression/linear_regression_main.cpp [hunk 1]
```diff
@@ -163,7 +163,7 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timer)
       timer.Start("load_responses");
       responses = params.Get<rowvec>("training_responses");
       timer.Stop("load_responses");
-
+
```
[the remark this answers is not in the document]
Comment (lumi232) [reply src/mlpack/methods/linear_regression/linear_regression_main.cpp:166]: Sure I'll remove this

File: src/mlpack/methods/linear_regression/linear_regression_impl.hpp [hunk 2]
```diff
@@ -297,22 +297,16 @@ LinearRegression<ModelMatType>::ComputeError(
   if (intercept)
   {
     // Ensure that we have the correct number of dimensions in the dataset.
-    if (nRows != parameters.n_rows - 1)
-    {
-      Log::Fatal << "The test data must have the same number of columns as the "
-          "training file." << std::endl;
-    }
+    util::CheckSameDimensionality(predictors, parameters.n_rows - 1,
+        "LinearRegression::ComputeError()", "predictors");
```
[the remark this answers is not in the document]
Comment (lumi232) [reply src/mlpack/methods/linear_regression/linear_regression_impl.hpp:301]: Sure I'll remove the nRows declaration.

File: src/mlpack/methods/linear_regression/linear_regression_main.cpp [hunk 3]
```diff
@@ -198,16 +198,14 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timer)
     mat points = std::move(params.Get<mat>("test"));
 
     // Ensure that test file data has the right number of features.
-    if ((lr->Parameters().n_elem - 1) != points.n_rows)
+    try
+    {
+      util::CheckSameDimensionality(points, lr->Parameters().n_elem - 1,
+        "Linear Regression Prediction", "test points");
```
[the remark this answers is not in the document]
Comment (lumi232) [reply src/mlpack/methods/linear_regression/linear_regression_main.cpp:204]: sure I'll remove this check

[the remark this answers is not in the document]
Comment (lumi232) [reply src/mlpack/methods/linear_regression/linear_regression_main.cpp]: done @rcurtin

File: src/mlpack/methods/linear_regression/linear_regression_impl.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (lumi232) [reply src/mlpack/methods/linear_regression/linear_regression_impl.hpp:300]: done @rcurtin

File: src/mlpack/methods/linear_regression/linear_regression_main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (lumi232) [reply src/mlpack/methods/linear_regression/linear_regression_main.cpp]: done @rcurtin

**Comment by lumi232** [comment]
I have made all the changes mentioned in the comments. The code is passing all the build tests (except windows). If there are any more changes do let me know. I'll move on to kmeans after this one is merged.

**Comment by lumi232** [comment]
Hey @rcurtin , can you please review this PR. I'll make the changes for other algorithms once this is merged so that we can close this issue(#2820) asap.

**Comment by lumi232** [comment]
@rcurtin can you review this PR.

File: src/mlpack/methods/linear_regression/linear_regression_main.cpp [hunk 4]
```diff
@@ -163,7 +163,6 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timer)
       timer.Start("load_responses");
       responses = params.Get<rowvec>("training_responses");
       timer.Stop("load_responses");
-
```
Comment (rcurtin) [inline src/mlpack/methods/linear_regression/linear_regression_main.cpp:166]: Sorry, what I meant was that there was no need to replace the blank line with a line that has four spaces.  I didn't mean that we should remove the line. :+1:

File: src/mlpack/methods/linear_regression/linear_regression_main.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (rcurtin) [reply src/mlpack/methods/linear_regression/linear_regression_main.cpp:203]: What did you try to debug the issue?  "The build tests failed" is not a sufficient reason to give up on the comment here.

**Pull request closed without merging**